File: lang-java/src/test/resources/fixtures/nested-lookup/negative/set-contains-in-loop.java
import java.util.List;
import java.util.Set;
import java.util.HashSet;

public class SetContainsInLoop {
    public void findMatches(List<String> items, Set<String> targetSet) {
        for (String item : items) {
            if (targetSet.contains(item)) {
                System.out.println(item);
            }
        }
    }
}

File: lang-kotlin/src/test/resources/fixtures/nested-lookup/negative/set-contains-in-loop.kt
class SetFilter {
    fun findMatches(items: List<String>, targetSet: Set<String>): List<String> {
        val result = mutableListOf<String>()
        for (item in items) {
            if (targetSet.contains(item)) {
                result.add(item)
            }
        }
        return result
    }
}
